code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -105,58 +117,116 @@ obj/machinery/atmospherics/proc/check_connect_types_construction(obj/machinery/a
-	if(!istype(icon_manager))
-		if(!safety) //to prevent infinite loops
-			icon_manager = new()
-			check_icon_cache(1)
-		return 0
+// Pipenet related functions	
+/obj/machinery/atmospherics/proc/returnPipenet()
+	return
 
-	return 1
+/obj/machinery/atmospherics/proc/returnPipenetAir()
+	return
 
-/obj/machinery/atmospherics/proc/color_cache_name(var/obj/machinery/atmospherics/node)
-	//Don't use this for standard pipes
-	if(!istype(node))
-		return null
+/obj/machinery/atmospherics/proc/setPipenet()
+	return
 
-	return node.pipe_color
-
-/obj/machinery/atmospherics/process()
-	build_network()
-
-/obj/machinery/atmospherics/proc/network_expand(datum/pipe_network/new_network, obj/machinery/atmospherics/pipe/reference)
-	// Check to see if should be added to network. Add self if so and adjust variables appropriately.
-	// Note don't forget to have neighbors look as well!
-
-	return null
+/obj/machinery/atmospherics/proc/replacePipenet()
+	return
 
 /obj/machinery/atmospherics/proc/build_network()
 	// Called to build a network from this node
-
-	return null
-
-/obj/machinery/atmospherics/proc/return_network(obj/machinery/atmospherics/reference)
-	// Returns pipe_network associated with connection to reference
-	// Notes: should create network if necessary
-	// Should never return null
-
-	return null
-
-/obj/machinery/atmospherics/proc/reassign_network(datum/pipe_network/old_network, datum/pipe_network/new_network)
-	// Used when two pipe_networks are combining
-
-/obj/machinery/atmospherics/proc/return_network_air(datum/network/reference)
-	// Return a list of gas_mixture(s) in the object
-	//		associated with reference pipe_network for use in rebuilding the networks gases list
-	// Is permitted to return null
+	return
 
 /obj/machinery/atmospherics/proc/disconnect(obj/machinery/atmospherics/reference)
+	return	
+	
+/obj/machinery/atmospherics/proc/nullifyPipenet(datum/pipeline/P)
+	if(P)
+		P.other_atmosmch -= src
+
+//(De)construction
+/obj/machinery/atmospherics/attackby(var/obj/item/weapon/W as obj, var/mob/user as mob)
+	if(can_unwrench && istype(W, /obj/item/weapon/wrench))
+		var/turf/T = get_turf(src)
+		if (level==1 && isturf(T) && T.intact)
+			user << "<span class='danger'>You must remove the plating first.</span>"
+			return 1
+		var/datum/gas_mixture/int_air = return_air()
+		var/datum/gas_mixture/env_air = loc.return_air()
+		add_fingerprint(user)
+		
+		var/unsafe_wrenching = FALSE
+		var/internal_pressure = int_air.return_pressure()-env_air.return_pressure()
+
+		playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
+		user << "<span class='notice'>You begin to unfasten \the [src]...</span>"
+		if (internal_pressure > 2*ONE_ATMOSPHERE)
+			user << "<span class='warning'>As you begin unwrenching \the [src] a gush of air blows in your face... maybe you should reconsider?</span>"
+			unsafe_wrenching = TRUE //Oh dear oh dear
+		
+		if (do_after(user, 40, target = src) && !gcDestroyed)

```

Comment:
I believe this should be isnull(gcDestroyed)...for whatever reason Krausus made our gcDestroyed different and you pretty much have to do the opposite of this (isValid and a lot of other things are also in line with this); you may want to look into GC code to confirm/deny.